experimental/devices/hd44780/lcd_hd44780_4bit.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// Package hd44780 controls the Hitachi LCD display chipset HD-44780
+//
+// Datasheet
+//
+// https://www.sparkfun.com/datasheets/LCD/HD44780.pdf
+package hd44780
+
+import (
+	"time"
+
+	"periph.io/x/periph/conn/gpio"
+)
+
+// LineTwo offset for the second line in the LCD buffer.
+const LineTwo = 0x40
+
+// LCD4 is the 4-bit addressing device for HD-44780
+type LCD4 struct {
+	// data pins
+	dataPins []gpio.PinOut
+
+	// register select pin
+	rsPin gpio.PinOut
+
+	// enable pin
+	enablePin gpio.PinOut
+}
+
+// NewLCD4Bit creates and initializes the LCD device
+//	data - references to data pins
+//	rs - rs pin
+//	e - strobe pin
+func NewLCD4Bit(data []gpio.PinOut, rs, e gpio.PinOut) (*LCD4, error) {
+	dev := LCD4{
```

Comment:
- you do not verify the length of data
- make dev a pointer right away, since it's meant to be returned